### PR TITLE
2023-06-06: bump up dependencies update identified by dependabot

### PR DESCRIPTION
**Pull request by chaochn47**
### Pure indirect dependency in [tools/mod/go.mod](https://github.com/etcd-io/etcd/blob/main/tools/mod/go.mod)
- [ ] https://github.com/etcd-io/etcd/pull/16009 should be closed. It is used by https://github.com/mgechev/revive/releases/tag/v1.3.2 which is the latest release tag. 
- [ ] https://github.com/etcd-io/etcd/pull/16010 should be closed. It is used by https://github.com/cloudflare/cfssl/releases/tag/v1.6.4 which is the latest release tag. 
- [ ] https://github.com/etcd-io/etcd/pull/16011 should be closed. It is used by https://github.com/mgechev/revive/releases/tag/v1.3.2 which is the latest release tag.

### Other indirect dependency
- [ ] https://github.com/etcd-io/etcd/pull/16012 should be closed. It is only used by `github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0` which uses `github.com/sirupsen/logrus@v1.4.2`. This version of indirect dependency seems not used by any modules. 

### Other dependency update
- [ ] https://github.com/etcd-io/etcd/pull/16013 should be closed and in favor of this PR to bump up version globally. 
- [ ] https://github.com/etcd-io/etcd/pull/16016 should be closed because `arduino/setup-protoc` v2.0.0 introduces a breaking change: https://github.com/arduino/setup-protoc/pull/78 and the release manager recommends projects that depend on old protoc version (like us v3.14.0) uses v1 in https://github.com/arduino/setup-protoc/pull/78#issuecomment-1568477933. Once https://github.com/etcd-io/etcd/issues/14533 is sorted out, we can bump up this dependency. 

### Test workflow passed and merged already
- [ ] https://github.com/etcd-io/etcd/pull/16014
- [ ] https://github.com/etcd-io/etcd/pull/16015

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks for keeping the docs up to date for the blocker with `setup-protoc`.

Note:
- `cfssl` has an open pr now to bump `go-sqlite3` so their next release should include that update.
- `go-grpc-middleware` have removed `github.com/sirupsen/logrus` as a depdendency in later releases so once we can update that the indirect for logrus should go away.

File: Documentation/contributor-guide/dependency_management.md [hunk 1]
```diff
@@ -76,7 +77,10 @@ For mixed cases, in which some modules directly while others indirectly depend o
 We should try to follow the first way, and temporarily fall back to the second one if we run into any issue on the first way. Eventually we
 should fix the issue and ensure all modules depend on the same version of the dependency.
 
-## About gRPC
+## Known incompatible dependency update
+- [build(deps): bump arduino/setup-protoc from 1.3.0 to 2.0.0](https://github.com/etcd-io/etcd/pull/16016)
+
+### About gRPC
```
Comment (ahrtr) [inline Documentation/contributor-guide/dependency_management.md:83]: Suggest to add them as two sub items under `Known incompatible dependency updates`.

## Known incomplete dependency updates
### arduino/setup-protoc
Please refer to [build(deps): bump arduino/setup-protoc from 1.3.0 to 2.0.0](https://github.com/etcd-io/etcd/pull/16016).

### About gRPC
......

Comment (chaochn47) [reply Documentation/contributor-guide/dependency_management.md:83]: I thought there might be a list of incompatible dependency updates. Using subtitle for a small item like `arduino/setup-protoc` may cause too many subtitles in the future. I don't think it's a big issue so I just move forward with your suggestion and we can always adjust the runbook later.

File: Documentation/contributor-guide/dependency_management.md [hunk 2]
```diff
@@ -37,14 +37,15 @@ in the following order,
 - go.etcd.io/etcd/tests/v3
 - go.etcd.io/etcd/v3
 - go.etcd.io/etcd/tools/v3
+For more details about etcd Golang modules, please check https://etcd.io/docs/v3.6/dev-internal/modules/
```
[the remark this answers is not in the document]
Comment (chaochn47) [reply Documentation/contributor-guide/dependency_management.md:40]: Thanks! TIL :p

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @chaochn47

**Pull request merged**